libs/features/mas/src/merch-card.js

```diff
@@ -372,6 +377,29 @@ export class MerchCard extends LitElement {
         for (const link of allLinks) {
             link.dataset.quantity = event.detail.option;
         }
+
+        if (this.prices.length === 0) return;
+
+        const price = this.prices[0];
+        if (this.quantityBasedPromo?.resolved && this.quantityBasedPromo?.promotionCode && price) {
+            if (event.detail.option >= this.quantityBasedPromo.minProductQuantity) {
```

Comment:
I think the whole promo min quantity logic should move in checkout-mixin.
I don't understand the intent here.  Promo code should remain on checkout links irrespective of the quantity.  It belongs checkout page to apply or not the promo based on the quantity.